cmd/go-filecoin/client.go

```diff
@@ -107,48 +114,101 @@ be 2, 1 hour would be 120, and 1 day would be 2880.
+		cmdkit.StringOption("peerid", "Override miner's peer id stored on chain"),
 	},
 	Run: func(req *cmds.Request, re cmds.ResponseEmitter, env cmds.Environment) error {
-		panic("not implemented pending full storage market integration")
+		addr, err := GetPorcelainAPI(env).WalletDefaultAddress()
+		if err != nil {
+			return err
+		}
 
-		//allowDuplicates, _ := req.Options["allow-duplicates"].(bool)
-		//
-		//miner, err := address.NewFromString(req.Arguments[0])
-		//if err != nil {
-		//	return err
-		//}
-		//
-		//data, err := cid.Decode(req.Arguments[1])
-		//if err != nil {
-		//	return err
-		//}
-		//
-		//askid, err := strconv.ParseUint(req.Arguments[2], 10, 64)
-		//if err != nil {
-		//	return err
-		//}
-		//
-		//duration, err := strconv.ParseUint(req.Arguments[3], 10, 64)
-		//if err != nil {
-		//	return err
-		//}
-		//
-		//resp, err := GetStorageAPI(env).ProposeStorageDeal()
-		//if err != nil {
-		//	return err
-		//}
-		//
-		//return re.Emit(resp)
+		maddr, err := address.NewFromString(req.Arguments[0])
+		if err != nil {
+			return err
+		}
+
+		chainHead := GetPorcelainAPI(env).ChainHeadKey()
+
+		dataCID, err := cid.Decode(req.Arguments[1])
+		if err != nil {
+			return errors.Wrap(err, "could not decode data cid")
+		}
+
+		data := &storagemarket.DataRef{
+			TransferType: "graphsync",
+			Root:         dataCID,
+		}
+
+		status, err := GetPorcelainAPI(env).MinerGetStatus(req.Context, maddr, chainHead)
+		if err != nil {
+			return err
+		}
+
+		peerID := status.PeerID
+		peerIDStr := req.Options["peerid"].(string)
+		if peerIDStr != "" {
+			peerID, err = p2pcore.Decode(peerIDStr)
+			if err != nil {
+				return err
+			}
+		}
+
+		providerInfo := &storagemarket.StorageProviderInfo{
+			Address:    maddr,
+			Owner:      status.OwnerAddress,
+			Worker:     status.WorkerAddress,
+			SectorSize: uint64(status.SectorSize),
+			PeerID:     peerID,
+		}
+
+		start, err := strconv.ParseUint(req.Arguments[2], 10, 64)
+		if err != nil {
+			return errors.Wrap(err, "could not parse deal start")
+		}
+
+		end, err := strconv.ParseUint(req.Arguments[3], 10, 64)
+		if err != nil {
+			return errors.Wrap(err, "could not parse deal end")
+		}
+
+		price, valid := types.NewAttoFILFromFILString(req.Arguments[4])
+		if !valid {
+			return errors.Errorf("could not parse price %s", req.Arguments[5])
+		}
+
+		collateral, valid := types.NewAttoFILFromFILString(req.Arguments[5])
+		if !valid {
+			return errors.Errorf("could not parse collateral %s", req.Arguments[6])
+		}
+
+		resp, err := GetStorageAPI(env).ProposeStorageDeal(
+			req.Context,
+			addr,
+			providerInfo,
+			data,
+			abi.ChainEpoch(start),
+			abi.ChainEpoch(end),
+			price,
+			collateral,
+			// proof version (not circuit or size) should be all that is important here
+			constants.DevRegisteredPoStProof,
```

Comment:
I am very surprised this is here at all. We should see what we can do to get rid of it (later). cc @hannahhoward